modules/repository-s3/src/yamlRestTest/java/org/elasticsearch/repositories/s3/RepositoryS3ClientYamlTestSuiteIT.java

```diff
@@ -8,20 +8,74 @@
 
 package org.elasticsearch.repositories.s3;
 
+import fixture.s3.S3HttpFixture;
+import fixture.s3.S3HttpFixtureWithEC2;
+import fixture.s3.S3HttpFixtureWithSessionToken;
+
 import com.carrotsearch.randomizedtesting.annotations.Name;
-import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakFilters;
 
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.LocalClusterSpecBuilder;
+import org.elasticsearch.test.fixtures.testcontainers.TestContainersThreadFilter;
 import org.elasticsearch.test.rest.yaml.ClientYamlTestCandidate;
-import org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase;
+import org.junit.ClassRule;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+import java.util.List;
+
+@ThreadLeakFilters(filters = { TestContainersThreadFilter.class })
+public class RepositoryS3ClientYamlTestSuiteIT extends AbstractRepositoryS3ClientYamlTestSuiteIT {
+
+    public static final S3HttpFixture s3Fixture = new S3HttpFixture(USE_FIXTURE);
+    public static final S3HttpFixtureWithSessionToken s3HttpFixtureWithSessionToken = new S3HttpFixtureWithSessionToken(USE_FIXTURE);
+    public static final S3HttpFixtureWithEC2 s3HttpFixtureWithEC2 = new S3HttpFixtureWithEC2(USE_FIXTURE);
+
+    private static final String s3TemporarySessionToken = "session_token";
+
+    @Override
+    protected List<String> blackListed() {
+        return USE_FIXTURE
+            ? List.of("repository_s3/50_repository_ecs_credentials/*", "repository_s3/60_repository_sts_credentials/*")
+            : List.of(
+                "repository_s3/30_repository_temporary_credentials/*",
+                "repository_s3/40_repository_ec2_credentials/*",
+                "repository_s3/50_repository_ecs_credentials/*",
+                "repository_s3/60_repository_sts_credentials/*"
+            );
+    }
+
+    public static ElasticsearchCluster cluster = configureCluster();
+
+    private static ElasticsearchCluster configureCluster() {
```

Comment:
nit: can we just inline this `configureCluster()` since it's not reused anywhere?

Reply:
If it's because of the conditional there are setting() and keystore() methods that accept a predicate.

Reply:
done. needed to add the same for systemProperty